src/controllers/libraryController.ts

```diff
@@ -31,36 +31,34 @@ export class LibraryController implements Disposable {
         this.disposable.dispose();
     }
 
-    public async addLibraries(libraryGlobs?: string[]) {
-        if (!libraryGlobs) {
-            libraryGlobs = [];
-            const workspaceFolder: WorkspaceFolder | undefined = Utility.getDefaultWorkspaceFolder();
-            const isWindows = process.platform.indexOf("win") === 0;
-            const results: Uri[] | undefined = await window.showOpenDialog({
-                defaultUri: workspaceFolder && workspaceFolder.uri,
-                canSelectFiles: true,
-                canSelectFolders: isWindows ? false : true,
-                canSelectMany: true,
-                openLabel: isWindows ? "Select jar files" : "Select jar files or directories",
-                filters: { Library: ["jar"] },
-            });
-            if (!results) {
-                return;
-            }
-            libraryGlobs = await Promise.all(results.map(async (uri: Uri) => {
-                // keep the param: `includeWorkspaceFolder` to false here
-                // since the multi-root is not supported well for invisible projects
-                const uriPath = workspace.asRelativePath(uri, false);
-                return (await fse.stat(uri.fsPath)).isDirectory() ? `${uriPath}/**/*.jar` : uriPath;
-            }));
-        }
-
+    public async addLibraryGlobs(libraryGlobs: string[]) {
         const setting = Settings.referencedLibraries();
         setting.exclude = this.dedupAlreadyCoveredPattern(libraryGlobs, ...setting.exclude);
         setting.include = this.updatePatternArray(setting.include, ...libraryGlobs);
         Settings.updateReferencedLibraries(setting);
     }
 
+    public async addLibraries(folders?: boolean) {
+        const workspaceFolder: WorkspaceFolder | undefined = Utility.getDefaultWorkspaceFolder();
+        const results: Uri[] | undefined = await window.showOpenDialog({
+            defaultUri: workspaceFolder && workspaceFolder.uri,
+            canSelectFiles: !folders,
+            canSelectFolders: folders,
```

Comment:
One possible regression here is: for the Mac users, they actually lose the capability to select both files and folders in one dialog.

So maybe still also check if it is in MacOS?

Reply:
I actually think it's better for the sake of simplicity to keep folder selection specific to the more advanced alt command. Since I don't think we can change the command title based on OS (without creating a third command that only displays on macOS), there is no consistent way to communicate with the user that *both* Jar files and folders can be selected when running macOS. However, I do agree that removing the capability is a regression (which might bother some users), so I added back folder selection to the `addLibraries` dialog on macOS in a735dc5.